android/pytorch_android/src/androidTest/java/org/pytorch/PytorchInstrumentedTests.java

```diff
@@ -298,6 +299,105 @@ public void testStr3Concat() throws IOException {
     }
   }
 
+  @Test
+  public void testAdd1ForByteTensors() throws IOException {
+    final long[] inputTensorShape = new long[]{256};
+    final long numElements = Tensor.numel(inputTensorShape);
+    final byte[] inputTensorData = new byte[(int) numElements];
+    for (int i = 0; i < numElements; ++i) {
+      inputTensorData[i] = (byte) i;
+    }
+    final Module module = Module.load(assetFilePath(TEST_MODULE_ASSET_NAME));
+
+    final Tensor inputTensorUint8 = Tensor.newUInt8Tensor(inputTensorShape, inputTensorData);
+    final Tensor inputTensorInt8 = Tensor.newInt8Tensor(inputTensorShape, inputTensorData);
+
+    final IValue inputUint8 = IValue.tensor(inputTensorUint8);
+    final IValue inputInt8 = IValue.tensor(inputTensorInt8);
+
+    final IValue outputUint8 = module.runMethod("tensorAdd1", inputUint8);
+    final Tensor outputTensorUint8 = outputUint8.getTensor();
+
+    final IValue outputInt8 = module.runMethod("tensorAdd1", inputInt8);
+    final Tensor outputTensorInt8 = outputInt8.getTensor();
+
+    byte[] outputDataUint8 = outputTensorUint8.getDataAsUnsignedByteArray();
+    byte[] outputDataInt8 = outputTensorInt8.getDataAsByteArray();
+
+    for (int i = 0; i < numElements; i++) {
+      final byte iB = (byte) i;
+      final byte outUint8B = outputDataUint8[i];
+      final byte outInt8B = outputDataInt8[i];
+
+      final int outUint8UI = byteToUnsignedInt(outUint8B);
+      final int expectedOutUint8UI = i != 255 ? i + 1 : 0;
+      final byte expectedOutUint8B = (byte) expectedOutUint8UI;
+      assertEqualsBytes(expectedOutUint8B, outUint8B);
+      assertEquals(expectedOutUint8UI, outUint8UI);
+
+      final int expectedOutInt8I = i == Byte.MAX_VALUE ? Byte.MIN_VALUE : ((int) iB) + 1;
+      final byte expectedOutInt8B = (byte) expectedOutInt8I;
+      assertEqualsBytes(expectedOutInt8B, outInt8B);
+      assertEquals(expectedOutInt8I, (int) outInt8B);
+    }
+  }
+
+  private static int byteToUnsignedInt(byte b) {
+    return ((int) b) & 0xff;
+  }
+
+  private static void assertEqualsBytes(byte expected, byte actual) {
+    if (expected != actual) {
+      assertTrue(String.format("Expected byte:%s(%s) actual byte:%s(%s)", Byte.toString(expected),
+          byteToBinaryString(expected), Byte.toString(actual), byteToBinaryString(actual)), false);
+    }
+  }
+
+  private static String byteToBinaryString(byte b) {
+    return String.format("%8s", Integer.toBinaryString(b & 0xFF)).replace(' ', '0');
```

Comment:
Does this work?

`return String.format("%08s", Integer.toBinaryString(b & 0xFF));`

Reply:
Fails :)
```
java.util.FormatFlagsConversionMismatchException: Conversion = s, Flags = 0
at java.util.Formatter$FormatSpecifier.failMismatch(Formatter.java:4131)
at java.util.Formatter$FormatSpecifier.checkBadFlags(Formatter.java:3035)
at java.util.Formatter$FormatSpecifier.checkGeneral(Formatter.java:2993)
at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2764)
at java.util.Formatter$FormatSpecifierParser.<init>(Formatter.java:2591)
at java.util.Formatter.parse(Formatter.java:2524)
at java.util.Formatter.format(Formatter.java:2472)
at java.util.Formatter.format(Formatter.java:2426)
at java.lang.String.format(String.java:2670)
at org.pytorch.PytorchInstrumentedTests.byteToBinaryString(PytorchInstrumentedTests.java:358)
```

Also tried `"%0$8s"`, does not fail, but does not fill with 0s `Expected byte:2(      10) actual byte:1(       1)`